system/jlib/jsocket.cpp

```diff
@@ -1585,15 +1585,15 @@ int CSocket::wait_read(unsigned timeout)
                 ret = -1;
             }
 #else
-            if (fds[0].revents & (POLLIN | POLLHUP))
+            if ( (fds[0].revents & (POLLERR | POLLNVAL)) || (!(fds[0].revents & (POLLIN | POLLHUP))) )
             {
```

Comment:
Yes, this is the reason for the order change.  Even if POLLIN is set we want to honor if POLLERR is also set.  Checking for POLLERR helps because after this there is a check for how many bytes are available to read and when its 0 the socket is closed and without the POLLERR check some error info could be missed.